### ISSUE TITLE
Get Identity on BulkMerge

### ISSUE THREAD

**Issue by kyletinsley** [zzzprojects/EntityFramework-Extensions#360]
Is there any way to have an identity value assigned when doing BulkMerge?  I've tried the following, but "Id" remains unchanged.  I'd love to not have to go back to the database to get the Id.

_db.BulkMerge(countries, options => options.ColumnOutputExpression = c => new { c.Id });

Country.Id [int, non-nullable, identity]

**Comment by JonathanMagnan** [comment]
Hello @kyletinsley ,

It should already return it by default, see the following example: https://dotnetfiddle.net/0Ba6ZB

Is it possible to provide a runnable project that should the current issue (identity not returned), we will look why in your case, it's not returned.

Best Regards,

Jon

**Comment by kyletinsley** [comment]
Thank you.  Yes, it does work, I tested the BulkMerge here:
[https://dotnetfiddle.net/Widget/kXkn3S]

I was using DataTableExtensions .AsEnumerable() on a DataTable to get my collection.  Then .Select() my POCOs from that for BulkMerge.  Apparently that creates an EnumerableRowCollection<DataRow>instead of the necessary IEnumerable<POCO>.  I've tacked on .ToList() and everything works perfectly.  Thanks for the awesome library!
